src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ServiceDescriptionService.java

```diff
@@ -534,6 +581,71 @@ public ServiceDescriptionType refreshServiceDescription(Long id, boolean ignoreW
         throw new NotImplementedException("REST ServiceDescription refresh not implemented yet");
     }
 
+    /**
+     * Refresh REST service description
+     *
+     * @param serviceDescriptionType
+     * @return {@link ServiceDescriptionType}
+     * @throws WrongServiceDescriptionTypeException if service type is not REST
+     * @throws ServiceNotFoundException             if there is no service linked to service description
+     * @throws InvalidUrlException                  if url is invalid
+     */
+    private ServiceDescriptionType refreshRESTServiceDescription(ServiceDescriptionType serviceDescriptionType)
+            throws WrongServiceDescriptionTypeException, ServiceNotFoundException, InvalidUrlException {
+        verifyAuthority("REFRESH_REST");
+
+        if (!serviceDescriptionType.getType().equals(DescriptionType.REST)) {
+            throw new WrongServiceDescriptionTypeException("Expected description type REST");
+        }
+
+        validateUrl(serviceDescriptionType.getUrl());
+
+        serviceDescriptionType.setRefreshedDate(new Date());
+
+        return serviceDescriptionType;
+    }
+
+    /**
+     * Refresh OPENAPI3 ServiceDescription
+     *
+     * @param serviceDescriptionType
+     * @param ignoreWarnings
+     * @return {@link ServiceDescriptionType}
+     * @throws WrongServiceDescriptionTypeException if service type is not openapi3
+     * @throws ServiceNotFoundException             if there is no service linked to given service description
+     * @throws UnhandledWarningsException           if unhandled warnings are found and ignoreWarnings if false
+     * @throws OpenApiParser.ParsingException       if parsing openapi3 description fails
+     * @throws InvalidUrlException                  if url is invalid
```

Comment:
`services.rb` does these checks

```
      check_duplicate_url(item)
      check_duplicate_service_codes(item)
```

refreshWSDLServiceDescription throws WsdlUrlAlreadyExistsException and ServiceAlreadyExistsException if this kind of checks fail. Are these missing from refreshOPENAPI3ServiceDescription?

Reply:
I think these checks are unnecessary for refreshOPENAPI3ServiceDescription as Service Description of type OPENAPI3 always contains just one service (servicecode and url) which won't change on refresh.